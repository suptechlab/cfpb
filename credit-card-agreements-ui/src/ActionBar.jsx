import './ActionBar.less';
import { changeSize, changeSort } from './actions/paging'
import { connect } from 'react-redux'
import { FormattedNumber } from 'react-intl'
import React from 'react';
import { showExportDialog } from './actions/dataExport'

const sizes = [ 10, 25, 50, 100 ]

/* eslint-disable camelcase */

const sorts = {
  created_date_desc: 'Sort by newest to oldest',
  created_date_asc: 'Sort by oldest to newest',
  relevance_desc: 'Sort by relevance',
  relevance_asc: 'Sort by relevance (asc)'
}

/* eslint-enable camelcase */

export class ActionBar extends React.Component {
  render() {
    // This is a hideous hack. When no filter is selected, the API returns
    // 20 results so we're checking for that quantity. Yes, if a series of
    // filter selections results in 20 results it will erroneously display
    // this message. We should instead check if filters are selected and if not,
    // display this message.
    if ( this.props.total === 20 ) {
      return (
        <summary className="action-bar">
          <div>
            <h4>Showing all credit cards</h4>
          </div>
        </summary>
      );
    }
    return (
        <summary className="action-bar">
          <div>
            <h4>
              Showing <FormattedNumber value={this.props.total} /> matched results
            </h4>
          </div>
        </summary>
    );
  }
}

export const mapStateToProps = state => ( {
  size: state.query.size,
  sort: state.query.sort,
  hits: state.results.total,
  total: state.results.doc_count
} )

export const mapDispatchToProps = dispatch => ( {
  onSize: ev => {
    const iSize = parseInt( ev.target.value, 10 )
    dispatch( changeSize( iSize ) )
  },
  onSort: ev => {
    dispatch( changeSort( ev.target.value ) )
  },
  onExportResults: () => {
    dispatch( showExportDialog() )
  }
} )

export default connect( mapStateToProps, mapDispatchToProps )( ActionBar )
