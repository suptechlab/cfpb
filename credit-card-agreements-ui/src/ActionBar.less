@import (less) "base.less";
@import (less) "select.less";

.action-bar {
  border: 1px solid @gray-40;
  border-left: 0;  // Don't double the border with the filters

  background-color: @green-20;

  .cf-select {
    padding-bottom: 0;
    border-left: 0;
    border-bottom: 0;

    select {
      padding-top: @gutter-minimum * 1.3;
      padding-left: @gutter-normal;

      background-color: transparent;
      font-family: @webfont-demi;
      font-size: 90%;
      text-transform: uppercase;
    }

    &:after {
      background-color: @green-20;
    }

    #choose-size {
      padding-left: @grid_gutter-width;
    }
  }

  h2 {
    padding: @gutter-normal @grid_gutter-width;
    margin: 0;

    @media @phone {
      padding: @gutter-normal;
    }
  }

  h4 {
    padding-left: @gutter-normal;
    padding-top: @gutter-minimum;
    margin: @gutter-normal;

    .a-btn {
      border-bottom: dotted 1px @dark-pacific;

      color: @dark-pacific;
      font-family: @webfont-demi;
      font-size: @size-v;
      text-transform: uppercase;

      &:hover {
        color: @link-text-hover;
      }
    }
  }
}
