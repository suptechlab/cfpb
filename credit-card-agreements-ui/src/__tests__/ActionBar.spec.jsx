import React from 'react';
import { IntlProvider } from 'react-intl';
import { ActionBar, mapDispatchToProps } from '../ActionBar';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <IntlProvider locale="en">
        <ActionBar total="100" hits="10" />
      </IntlProvider>
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it('hooks into onSize', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).onSize({target: { value: '50' }});
      expect(dispatch.mock.calls.length).toEqual(1);
    })

    it('hooks into onSort', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).onSort({target: { value: 'foo' }});
      expect(dispatch.mock.calls.length).toEqual(1);
    })

    it('hooks into onExportResults', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).onExportResults();
      expect(dispatch.mock.calls.length).toEqual(1);
    })
  })
});

