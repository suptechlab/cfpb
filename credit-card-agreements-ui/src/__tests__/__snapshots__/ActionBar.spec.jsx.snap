// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders without crashing 1`] = `
<summary
  className="action-bar"
>
  <div>
    <h4>
      Showing 
      <span>
        100
      </span>
       matched results
    </h4>
  </div>
</summary>
`;
